File: problems/problems_LCP_40/solution.ts
function maxmiumScore(cards: number[], cnt: number): number {
	cards.sort((a, b) => b - a);
	let sum: number = 0;
	for (let i: number = 0; i < cnt; i++) {
		sum += cards[i];
	}
	if (sum % 2 === 0) {
		return sum;
	}
	const cur: number = cards[cnt - 1];
	const replaceSum: Function = (x: number): number => {
		for (let i: number = cnt; i < cards.length; i++) {
			if (cards[i] % 2 !== x % 2) {
				return sum - x + cards[i];
			}
		}
		return 0;
	}
	let ans: number = replaceSum(cur);
	for (let i: number = cnt - 2; i >= 0; i--) {
		if (cards[i] % 2 !== cur % 2) {
			ans = Math.max(ans, replaceSum(cards[i]));
			break;
		}
	}
	return ans;
};

export function Solve(inputJsonElement: string): any {
	const inputValues: string[] = inputJsonElement.split("\n");
	const cards: number[] = JSON.parse(inputValues[0]);
	const cnt: number = JSON.parse(inputValues[1]);
	return maxmiumScore(cards, cnt);
}

File: problems/problems_LCP_40/solution.rs
use serde_json::{json, Value};

pub struct Solution;

impl Solution {
    pub fn maxmium_score(cards: Vec<i32>, cnt: i32) -> i32 {
		let mut cards = cards;
		cards.sort_unstable_by(|a, b| b.cmp(a));
		let cnt = cnt as usize;
		let s: i32 = cards.iter().take(cnt).sum();
		if s % 2 == 0 {
			return s;
		}
		let cur = cards[cnt - 1];
		let replace_sum = |x: i32| -> i32 {
			for &v in cards.iter().skip(cnt) {
				if v % 2 != x % 2 {
					return s - x + v;
				}
			}
			0
		};
		let mut ans = replace_sum(cur);
		for &v in cards[..cnt - 1].iter().rev() {
			if v % 2 != cur % 2 {
				ans = ans.max(replace_sum(v));
				break;
			}
		}
		ans
    }
}

#[cfg(feature = "solution_LCP_40")]
pub fn solve(input_string: String) -> Value {
	let input_values: Vec<String> = input_string.split('\n').map(|x| x.to_string()).collect();
	let cards: Vec<i32> = serde_json::from_str(&input_values[0]).expect("Failed to parse input");
	let cnt: i32 = serde_json::from_str(&input_values[1]).expect("Failed to parse input");
	json!(Solution::maxmium_score(cards, cnt))
}
